code/go/0chain.net/chaincore/chain/entity.go

```diff
@@ -693,6 +693,9 @@ func (c *Chain) addBlock(b *block.Block) *block.Block {
 		return eb
 	}
 	c.blocks[b.Hash] = b
+
+	c.pushBlockEvent(b)
```

Comment:
We only save finalized blocks to event db, so perhaps do it in https://github.com/0chain/0chain/blob/9b72325418072680429bc00b654367b1234b6af9/code/go/0chain.net/chaincore/chain/protocol_block.go#L351.

Reply:
I've done it intentionally, we might need all proposed blocks here, not only finalized blocks, at least all notarized so I suggest to save them here